controllers/gce/loadbalancers/loadbalancers.go

```diff
@@ -88,19 +88,8 @@ func NewLoadBalancerPool(
 }
 
 func (l *L7s) create(ri *L7RuntimeInfo) (*L7, error) {
-	// Lazily create a default backend so we don't tax users who don't care
-	// about Ingress by consuming 1 of their 3 GCE BackendServices. This
-	// BackendService is deleted when there are no more Ingresses, either
-	// through Sync or Shutdown.
 	if l.glbcDefaultBackend == nil {
-		err := l.defaultBackendPool.Add(l.defaultBackendNodePort)
-		if err != nil {
-			return nil, err
-		}
-		l.glbcDefaultBackend, err = l.defaultBackendPool.Get(l.defaultBackendNodePort)
-		if err != nil {
-			return nil, err
-		}
+		glog.Warningf("Creating l7 without a default backend")
```

Comment:
we were always creating it as part of sync: https://github.com/kubernetes/contrib/blob/master/ingress/controllers/gce/loadbalancers/loadbalancers.go#L180, so this was just a redundant check. But we were also deleting it as part of sync, when there are no more lbs: https://github.com/kubernetes/contrib/blob/master/ingress/controllers/gce/loadbalancers/loadbalancers.go#L191

That's not the right time to try and delete it, because a url-map could still be referencing it (the delete would fail saying: "backend already in use", then we'd requeue, come back in, delete the url map, and now the backend delete would pass). 

So I moved the delete to GC(), which happens after url map deletion. 
https://github.com/kubernetes/ingress/pull/21/files#diff-247e0d4a5e00d30d32c0e454986614cdR210